tools/importer-rest-api-specs/components/testing/complete.go

```diff
@@ -13,6 +13,19 @@ func (tb TestBuilder) generateCompleteTest(dependencies *testDependencies) (*str
 	if !ok {
 		return nil, fmt.Errorf("the schema model %q was not found", tb.details.SchemaModelName)
 	}
+
+	completeRequired := false
+	for _, details := range topLevelModel.Fields {
+		if details.Required == false {
+			completeRequired = true
+			break
+		}
+	}
+
+	if !completeRequired {
```

Comment:
Returning `nil, nil` here since in `generate.go` we have checks to omit generation of the complete test if this function returns nil: 

https://github.com/hashicorp/pandora/blob/a608cd5ff33fb038760d51d246c8d917779f02cc/tools/importer-rest-api-specs/components/testing/generate.go#L60-L63

However this unit test fails expects that if there are no properties that we generate an empty config which I don't think is right. Is there a reason that we've written this test case like so?
https://github.com/hashicorp/pandora/blob/a608cd5ff33fb038760d51d246c8d917779f02cc/tools/importer-rest-api-specs/components/testing/complete_test.go#L13

Reply:
> Is there a reason that we've written this test case like so?

Looking at the code I'm gonna wager the Basic tests were copied and then updated to match - I don't think there's a need to support that scenario now since we're filtering the tests out - so we should be good to remove the test 👍

It's worth noting (given the name) that there _is_ a scenario where the `properties` object is required to be sent (but doesn't contain any fields - since else the API crashes [we've seen this a handful of times]) - but that's not what this test is testing, so it's fine 👍